test/Unit/email/address-test.php

```diff
@@ -0,0 +1,37 @@
+<?php
+declare(strict_types=1);
+
+namespace email;
+
+use jsys\types\email\Address;
+use PHPUnit\Framework\TestCase;
+
+
+class AddressTest extends TestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp(); // TODO: Change the autogenerated stub
+    }
+
+    public function test_must_not_initialize_with_invalid_email()
+    {
+        $testInvalidEmail = 'hussain@gmail';
+        $this->expectExceptionCode(1001);
+        $this->expectExceptionMessage("Given string '$testInvalidEmail' is invalid.");
+        $emailAddress = new Address($testInvalidEmail);
+    }
+
+    public function test_can_initialize_with_valid_email()
+    {
```

Comment:
add few valid email variations

Reply:
Would you please give me an example for valid data provider.

Reply:
abc@domain.com
abc.com@kanji.com
abc-roze@site.co
someName@myhome.org
1234@allnumbers.jp
1234-text@allnumbers.jp

Reply:
Sorry sir, I wanted to ask about how can I implement this?
Would you send me some code example?